bottlecap/src/lifecycle/invocation/triggers/step_function_event.rs

```diff
@@ -118,16 +131,36 @@ impl Trigger for StepFunctionEvent {
 impl StepFunctionEvent {
     #[must_use]
     pub fn get_span_context(&self) -> SpanContext {
-        let (lo_tid, hi_tid) = Self::generate_trace_id(self.execution.id.clone());
-        let tags = HashMap::from([(
-            DATADOG_HIGHER_ORDER_TRACE_ID_BITS_KEY.to_string(),
-            format!("{hi_tid:x}"),
-        )]);
+        let (lo_tid, tags) =
+            if let (Some(trace_id), Some(trace_tags)) = (&self.trace_id, &self.trace_tags) {
```

Comment:
We wouldn't need this if-else checking if we defined separate events rather than including all fields in one and making some optional

**But**, this way we're keeping all the complexity in this file, otherwise we'd have to match on all the different types of events in the `span_inferrer`

This logic will need infrequent edits so it made sense to me to pull it down similar to what we do in the python layer